clang-tools-extra/clang-tidy/readability/UseStdMinMaxCheck.cpp

```diff
@@ -0,0 +1,189 @@
+//===--- UseStdMinMaxCheck.cpp - clang-tidy -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "UseStdMinMaxCheck.h"
+#include "../utils/ASTUtils.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include "clang/Lex/Preprocessor.h"
+#include <iostream>
+using namespace std;
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::readability {
+
+class ExprVisitor : public clang::RecursiveASTVisitor<ExprVisitor> {
+public:
+  explicit ExprVisitor(clang::ASTContext *Context) : Context(Context) {}
+  bool visitStmt(const clang::Stmt *S, bool &found,
+                 clang::QualType &GlobalImplicitCastType) {
+
+    if (isa<clang::ImplicitCastExpr>(S) && !found) {
+      found = true;
+      const clang::ImplicitCastExpr *ImplicitCast =
+          cast<clang::ImplicitCastExpr>(S);
+      GlobalImplicitCastType = ImplicitCast->getType();
+      // Stop visiting children.
+      return false;
+    }
+    // Continue visiting children.
+    for (const clang::Stmt *Child : S->children()) {
+      if (Child) {
+        this->visitStmt(Child, found, GlobalImplicitCastType);
+      }
+    }
+
+    return true; // Continue visiting other nodes.
+  }
+
+private:
+  clang::ASTContext *Context;
```

Comment:
This variable is unused, please remove it

Reply:
Not able to remove it , actually its been used, sorry If I misunderstood anything can you please explain it.

Reply:
![Screenshot from 2024-01-27 23-13-22](https://github.com/llvm/llvm-project/assets/76656712/09978a9d-e45b-4a24-b296-a3c71ec47640)
In the definition, Context is used in the constructor as if it were a member of the class, so if it's removed it will not declared anywhere in the class.

Reply:
I can see that, but you're not using that variable anywhere. What's the point of declaring it if you're only initializing it and not using it?

Reply:
yes yes , when I remove that private context its showing error, sorry if I am not able to get what you are trying to say , How can I ensure to not get error?

Reply:
I get it , sorry for my previous silly comments, I removed the initialisation